java-checks/src/main/java/org/sonar/java/checks/security/EncryptionAlgorithmCheck.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.ExpressionsHelper;
+import org.sonar.java.checks.helpers.JavaPropertiesHelper;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.TypeCriteria;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+
+@Rule(key = "S5542")
+public class EncryptionAlgorithmCheck extends AbstractMethodDetection {
+
+  private static final Pattern ALGORITHM_PATTERN = Pattern.compile("(.+)/(.+)/(.+)");
+
+  @Override
+  protected List<MethodMatcher> getMethodInvocationMatchers() {
+    return Collections.singletonList(
+      MethodMatcher.create()
+        .typeDefinition(TypeCriteria.is("javax.crypto.Cipher"))
+        .name("getInstance")
+        .withAnyParameters());
+  }
+
+  @Override
+  protected void onMethodInvocationFound(MethodInvocationTree mit) {
+    if (mit.arguments().isEmpty()) {
+      return;
+    }
+    ExpressionTree firstArgument = mit.arguments().get(0);
+    ExpressionTree defaultPropertyValue = JavaPropertiesHelper.retrievedPropertyDefaultValue(firstArgument);
+    ExpressionTree algorithmTree = defaultPropertyValue == null ? firstArgument : defaultPropertyValue;
+    String algorithmName = ExpressionsHelper.getConstantValueAsString(algorithmTree).value();
+    if (algorithmName != null && isInsecureAlgorithm(algorithmName)) {
+      reportIssue(firstArgument, "Use secure mode and padding scheme.");
+    }
+  }
+
+  private static boolean isInsecureAlgorithm(String algorithmName) {
+    Matcher matcher = ALGORITHM_PATTERN.matcher(algorithmName);
+    if (matcher.find()) {
+      String algorithm = matcher.group(1);
+      String mode = matcher.group(2);
+      String padding = matcher.group(3);
+
+      if (mode.equals("ECB")) {
```

Comment:
for all these string comparison, please invert the string constant with the variable. See rule [S1132](https://next.sonarqube.com/sonarqube/coding_rules?open=java%3AS1132&rule_key=java%3AS1132).